python/pyarrow/table.pxi

```diff
@@ -2453,6 +2453,49 @@ cdef class Table(_PandasConvertible):
 
         return table
 
+    def join(self, keys, right_table, right_keys=None, join_type="left outer",
```

Comment:
We should maybe switch the order of the `keys` and `right_table` arguments. All the other dataframe libraries I was checking have the other table as first argument

Reply:
done